### ISSUE TITLE
mod.json dependencies loaded out of order

### ISSUE THREAD

**Issue by Powback** [VeniceUnleashed/VeniceUnleashed#494]
modlist.txt
```
VEDemo
VEManager
```
Note: VEManager is loaded first if the dependency field is removed, meaning that modlist.txt is processed backwards?

VEDemo mod.json
```
{
	"Name": "VEDemo",
	"Authors": ["Powback"],
	"Description": "Visual Environments yo",
	"URL": "",
	"Version": "1.0",
	"HasWebUI": false,
	"HasVeniceEXT": true,
	"Dependencies": {
		"VEManager": "1.0"
	}
}

```

VEManager mod.json
```
{
	"Name": "VEManager",
	"Authors": ["Powback"],
	"Description": "Handles VE",
	"URL": "",
	"Version": "1.0",
	"HasWebUI": false,
	"HasVeniceEXT": true
}
```

Server output:
```
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] [vedemo] Compiling client module...
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] [vedemo] Compiling client script '__init__.lua'...
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] Successfully loaded VeniceEXT module 'vedemo'.
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] Loading VeniceEXT module 'vemanager'.
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] [vemanager] Compiling client module...
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] [vemanager] Compiling client script 'easing.lua'...
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] [vemanager] Compiling client script 've_base.lua'...
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] [vemanager] Compiling client script 've_preset.lua'...
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] [vemanager] Compiling client script '__init__.lua'...
[2020-06-22 15:26:56+02:00] [info] [VeniceEXT] Successfully loaded VeniceEXT module 'vemanager'.
```

Client output:
```
[2020-06-22 15:21:10+02:00] [info] [VeniceEXT] Loading VeniceEXT module 'vedemo'.
[2020-06-22 15:21:10+02:00] [info] [VeniceEXT] Successfully loaded VeniceEXT module 'vedemo'.
[2020-06-22 15:21:10+02:00] [info] [VeniceEXT] Loading VeniceEXT module 'vemanager'.
[2020-06-22 15:21:10+02:00] [info] [VeniceEXT] [vemanager] Initializing VEManagerClient
[2020-06-22 15:21:10+02:00] [info] [VeniceEXT] Successfully loaded VeniceEXT module 'vemanager'.
```

This results in VEDemo having events such as LevelLoaded called before VEManager does. Logically dependencies should be initialized before the dependee to ensure the correct order of execution.

Removing the dependency field allows 

TL;DR:
Dependencies are loaded after the mod, the dependency should be loaded first no?

**Comment by OrfeasZ** [comment]
Fixed in build `16841`.
